debugoverlay-ext-netstats/src/main/java/com/ms_square/debugoverlay_ext_netstats/NetStatsDataModule.java

```diff
@@ -0,0 +1,82 @@
+package com.ms_square.debugoverlay_ext_netstats;
+
+import android.net.TrafficStats;
+import android.os.Handler;
+import android.os.Looper;
+
+import com.ms_square.debugoverlay.modules.BaseDataModule;
+
+import java.util.Locale;
+
+public class NetStatsDataModule extends BaseDataModule<String> {
+
+    private static final String TAG = NetStatsDataModule.class.getSimpleName();
+
+    private static final String HEADER = "Received: %8s/s\nSent: %12s/s";
+
+    private final Handler handler = new Handler(Looper.getMainLooper());
+
+    private final int intervalMilliseconds;
+    private final int uid;
+    private final double intervalSeconds;
+
+    private Double previousReceived = null;
```

Comment:
instead of using primitive wrapper object, could you just use primitives? I basically want to avoid unnecessary object creations.
You could initialize it to -1f, and then only calculate your stats values when it is >= 0.

Reply:
``TrafficStats.getUid*xBytes`` returns ``TrafficStats.UNSUPPORTED`` upon failure, so turning the ``previous*`` variables into primitives and initializing them to  -1 required some additional changes. Let me know what you think of this approach.